contrib/gocql/gocql/gocql.go

```diff
@@ -111,6 +111,17 @@ func (tq *Query) newChildSpan(ctx context.Context) ddtrace.Span {
 }
 
 func (tq *Query) finishSpan(span ddtrace.Span, err error) {
+	if err == gocql.ErrNotFound {
```

Comment:
> implement this in the WithErrorCheck fashion

Can you clarify? I think you are asking us to examine `gorm.WithErrorCheck` (an option for `gorm` which we do not use) and we should create a new function `tracer.WithErrorCheck` to have behavior like `gorm.WithErrorCheck`, and alter the logic of trace span ending to accommodate. Basically implement PR 806 but for trace spans.
https://github.com/DataDog/dd-trace-go/pull/806/files
https://github.com/DataDog/dd-trace-go/blob/main/ddtrace/tracer/option.go#L829-L836

> ensure we don't break anyone.

Can you give examples of how this would break existing code? This PR only alters the error passed to the _trace span_. Callers to the gocql function are not affected.